Makefile

```diff
@@ -317,6 +320,16 @@ $(ROMC): $(ROM) $(ELF) $(BUILD_DIR)/compress_ranges.txt
 $(ELF): $(TEXTURE_FILES_OUT) $(ASSET_FILES_OUT) $(O_FILES) $(OVL_RELOC_FILES) $(BUILD_DIR)/ldscript.txt $(BUILD_DIR)/undefined_syms.txt
 	$(LD) -T $(BUILD_DIR)/undefined_syms.txt -T $(BUILD_DIR)/ldscript.txt --no-check-sections --accept-unknown-input-arch --emit-relocs -Map $(BUILD_DIR)/z64.map -o $@
 
+# Python virtual environment
+
+$(VENV):
+	python3 -m venv $(VENV)
+
+$(VENV)/.install-python-deps: requirements.txt | $(VENV)
```

Comment:
tbh I don't really see the need to separate this out, since if the venv already exists `python3 -m venv $(VENV)` does nothing so is fine with the others.

Reply:
I think `python3 -m venv $(VENV)` will reinstall the virtualenv, at least I tried editing `.venv/pyvenv.cfg` and ran `python3 -m venv $(VENV)` and it was changed back. It also took about 2 seconds to do so which is pretty slow. I guess the 2 seconds isn't so bad since it will only run when requirements.txt changes, but maybe advanced users want to set up their virtualenv differently (using a different python3 perhaps) and I don't think we should clobber that

Reply:
Hmm, it does seem to clobber `.venv/pyvenv.cfg` but seems to keep pip packages installed.
Perhaps we just go back to using `test -d $(VENV) || python3 -m venv $(VENV)` to see if it already exists. Would kind of like to keep one target as much as possible.

Reply:
Though if we do end up keeping both targets might I suggest `venv` and `venv-deps` as the target names.